# TODO
# [1/2] –î–æ–¥–µ–ª–∞—Ç—å –ø—Ä–∏—Ü–µ–ª—ã
# [*] –°–¥–µ–ª–∞—Ç—å –∫–Ω–æ–ø–∫–∏ –æ—Ç–∫—Ä—ã—Ç–∏—è —Å–∞–π—Ç–∞ –º–æ–¥–∞ –≤ —Å–ø–∏—Å–∫–µ –º–æ–¥–æ–≤
# [*] –î–æ–±–∞–≤–∏—Ç—å –±–æ–ª—å—à–µ —Ä–µ–∂–∏–º–æ–≤ –≤ –∏–≥—Ä—É
# [1/4] –î–æ–¥–µ–ª–∞—Ç—å –º–∞–≥–∞–∑–∏–Ω –∏ –∏–Ω–≤–µ–Ω—Ç–∞—Ä—å (—Å–∫–ª–∞–¥)
# [*] –î–æ–±–∞–≤–∏—Ç—å —Ä–µ–∂–∏–º—ã –≤ –º–æ–¥–ª–æ–∞–¥–µ—Ä
# [-] –í–µ—Ä–Ω—É—Ç—å Discord Rich Preference (–∏–ª–∏ –∫–∞–∫ –µ–≥–æ) –∏–∑ 1.3 –∏ –Ω–∏–∂–µ
# [*] –¥–æ–±–∞–≤—å—Ç–µ –±–æ–ª—å—à–µ TODOü§™ü§™ü§™ 
# [ ] –¥–æ–±–∞–≤–∏—Ç—å –∞–≤—Ç–æ–º–æ–±–∏–ª—å–Ω—ã–π —Ä–µ–∂–∏–º
# [ ] —Å–¥–µ–ª–∞—Ç—å –ï–©–Å –±–æ–ª—å—à–µ TODO tm

# INFO —Å–∞–º—ã–µ –≤–∞–∂–Ω—ã–µ –ø–µ—Å–Ω–∏ –∑—Ä 2.0 —ç—Ç–æ:
# –±—É—Ç—ã—Ä–∫–∞ –º–µ—Ç–µ–æ—Ä–∏—Ç

# —á—Ç–æ–æ–æ –≥–æ–¥–æ—Ç –ø–æ–¥—Å–≤–µ—Ç–∫–∞ –∫–æ–º–º–µ–Ω—Ç–∞—Ä–∏–µ–≤üò®
# ALERT, ATTENTION, CAUTION, CRITICAL, DANGER, SECURITY
# BUG, DEPRECATED, FIXME, HACK, TASK, TBD, TODO, WARNING
# INFO, NOTE, NOTICE, TEST, TESTING

extends Node

# –ù–∞—Å—Ç—Ä–æ–π–∫–∏ –∏ —Å–ª—É–∂–µ–±–Ω–æ–µ
var VERSION = ProjectSettings.get_setting("application/config/version")
var FULLSCREEN = false
var SmoothTransitions = false
var WEAPONHINTS = true
@onready var GAME = "res://gamemode.tscn"
@onready var SETTINGS = "res://settings.tscn"
const isDEMO = true # –î–∞–Ω–Ω–∞—è –Ω–∞—Å—Ç—Ä–æ–π–∫–∞ –æ—Ç–∫–ª—é—á–∞–µ—Ç –º–∞–≥–∞–∑–∏–Ω, —Å–∫–ª–∞–¥ –∏ —Å–ø–∏—Å–æ–∫ –º–æ–¥–æ–≤ –û–Ω–ª–∞–π–Ω, —Ç–∞–∫-–∫–∞–∫ –æ–Ω–æ –Ω–µ –≥–æ—Ç–æ–≤–æ (–æ–Ω–ª–∞–π–Ω –º–æ–¥—ã —è –µ—â—ë –∞–ø–∏ –Ω–µ —Å–¥–µ–ª–∞–ª –Ω—É —è –∏ –ª–æ—Ö –≤–æ–æ–±—â–µ)

#–ö–æ–Ω—Ñ–∏–≥–∏
const SAVE_PATH = "user://save.cfg"
var CONFIG = ConfigFile.new()
var KT_URL = "https://kteam.veliona.no/"

# –ü–µ—Ä–µ–º–µ–Ω–Ω—ã–µ
var BGID
var FROM = 0
var ZCOINS = 0
var CURRENT_AIM = preload("res://Resources/aims/default.png")
var GAYPAD_AIDS = 1 # –ø—Ä–∏—á–µ–º —Ç—É—Ç aids? –Ω—É, —Ç–∏–ø–∞ –∫–æ—Ä–æ—á–µ —Ç–∏–ø–∞ —è —Ö–æ—Ç–µ–ª —Å–Ω–∞—á–∞–ª–∞ gaypad_speed, –Ω—É —Ç–∏–ø–∞ —Å–ø–∏–¥, –∞ —Å–ø–∏–¥ –ø–æ –∞–Ω–≥–ª–∏–π—Å–∫–∏ —ç—Ç–æ aids –≤–æ—Ç —Ç–∏–ø–∞ —à—É—Ç–∫–∞ –¥–∞ —Ö–∞–∑–≤—ä–∑–∞—Ö–≤–∑–∞—Ö–≤–∑–∞—Ö–≤–∑–∞

func _input(event: InputEvent) -> void:
	if event.is_action_pressed("fullscreenkey"):
		if FULLSCREEN == false:
			DisplayServer.window_set_mode(DisplayServer.WINDOW_MODE_FULLSCREEN)
			FULLSCREEN = true
		else:
			DisplayServer.window_set_mode(DisplayServer.WINDOW_MODE_WINDOWED)
			FULLSCREEN = false
	
func _ready() -> void:
	CONFIG.load(SAVE_PATH)
	if !CONFIG.get_value("settings", "lang"):
		TranslationServer.set_locale("en")
	else:
		TranslationServer.set_locale(CONFIG.get_value("settings", "lang"))
		
	if CONFIG.get_value("settings", "smoothtransitions"):
		SmoothTransitions = CONFIG.get_value("settings", "smoothtransitions")
	
	if CONFIG.get_value("save", "disableweaponhints"):
		WEAPONHINTS = CONFIG.get_value("save", "disableweaponhints")
		
	if CONFIG.get_value("settings", "fullscreen"):
		DisplayServer.window_set_mode(DisplayServer.WINDOW_MODE_FULLSCREEN)
		FULLSCREEN = CONFIG.get_value("settings", "fullscreen")
		
	if CONFIG.get_value("settings", "gamepad_sens"):
		GAYPAD_AIDS = CONFIG.get_value("settings", "gamepad_sens")
		
	if CONFIG.get_value("save", "zcoins"):
		ZCOINS = CONFIG.get_value("save", "zcoins")
		

		
	# tf2 reference ALERT
	if !FileAccess.file_exists("res://_IMPORTANT_IMAGE_DONT_DELETE_INACHE_PISEC!!.jpg"):
		OS.alert("–ò–≥—Ä–∞ –æ–±–Ω–∞—Ä—É–∂–∏–ª–∞ –æ—Ç—Å—É—Ç—Å—Ç–≤–∏–µ —Å–∞–º–æ–≥–æ –≤–∞–∂–Ω–æ–≥–æ —Ñ–∞–π–ª–∞. –ò–≥—Ä–∞ –±–æ–ª—å—à–µ –Ω–µ –∑–∞–ø—É—Å—Ç–∏—Ç—Å—è. –ù–∞–≤–µ—Ä–Ω–æ–µ.\n–í–µ—Ä–Ω–∏—Ç–µ —Ñ–∞–π–ª –Ω–∞ –º–µ—Å—Ç–æ –∏ –Ω–µ —Ç—Ä–æ–≥–∞–π—Ç–µ –µ–≥–æ –µ–º—É —Å—Ç—Ä–∞—à–Ω–æ –≤–æ–æ–±—â–µ –∫–∞–ø–µ—Ü.", "Aw, snap")
		OS.crash("Fatal error")
		

# –ø–æ–ª–µ–∑–Ω–∞—è —Ñ—É–Ω–∫—Ü–∏—è‚Ñ¢
func check(–≤_—Ä–æ—Ç_–º–Ω–µ_–Ω–æ–≥–∏: bool):
	if –≤_—Ä–æ—Ç_–º–Ω–µ_–Ω–æ–≥–∏:
		return true
	else:
		return false
		
func got_finishedsign(value):
	# ID –∫–æ—Ç–æ—Ä–æ–µ –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è –¥–ª—è –æ–ø—Ä–µ–¥–µ–ª–µ–Ω–∏—è, –≤ –∫–∞–∫—É—é —Å—Ü–µ–Ω—É –æ—Ç–ø—Ä–∞–≤–∏—Ç—å –ø–æ—Å–ª–µ –∞–Ω–∏–º–∞—Ü–∏–∏. –ü–æ–∫–∞-—á—Ç–æ —Ä–∞–±–æ—Ç–∞—é—Ç —Ç–æ–ª—å–∫–æ –≤—Å—Ç—Ä–æ–µ–Ω–Ω—ã–µ –∞–Ω–∏–º–∞—Ü–∏–∏/—Å—Ü–µ–Ω—ã.
	match value:
		1:
			await get_tree().process_frame
			get_tree().change_scene_to_file(GAME)
		2:
			await get_tree().process_frame
			get_tree().change_scene_to_file(SETTINGS)
		3:
			await get_tree().process_frame
			get_tree().change_scene_to_file("res://mods.tscn")
		4:
			
			get_tree().quit()
		5:
			await get_tree().process_frame
			get_tree().change_scene_to_file("res://modsinternet.tscn")
		6:
			await get_tree().process_frame
			get_tree().change_scene_to_file("res://menu.tscn")
